src/ewatercycle/_forcings/caravan.py

```diff
@@ -0,0 +1,240 @@
+                 'aridity',
+                 'frac_snow',
+                 'moisture_index',
+                 'seasonality',
+                 'high_prec_freq',
+                 'high_prec_dur',
+                 'low_prec_freq',
+                 'low_prec_dur']
+
+RENAME_ERA5 = {'total_precipitation_sum':'pr',
+                'potential_evaporation_sum':'evspsblpot',
+                 'temperature_2m_mean': 'tas',
+                 'temperature_2m_min': 'tasmin',
+                 'temperature_2m_max': 'tasmax',
+                 'streamflow':'Q'}
+class Caravan(DefaultForcing):
+    """Retrieves specified part of the caravan dataset from the OpenDAP server.
+
+    Examples:
+
+    The caravan dataset is an already prepared set by Frederik Kratzert,
+    (see https://doi.org/10.1038/s41597-023-01975-w).
+
+    This retrieves it from the OpenDAP server of 4TU,
+    (see https://doi.org/10.4121/bf0eaf7c-f2fa-46f6-b8cd-77ad939dd350.v4).
+
+    This can be done by specifying the
+
+    .. code-block:: python
+
+        from pathlib import Path
+        from rich import print
+        from ewatercycle.forcing import sources
+
+        path = Path.cwd()
+        forcing_path = path / "Forcing"
+        experiment_start_date = "1997-08-01T00:00:00Z"
+        experiment_end_date = "2005-09-01T00:00:00Z"
+        HRU_id = 1022500
+
+        camels_forcing = sources['LumpedCaravanForcing'].retrieve(start_time = experiment_start_date,
+                                                                  end_time = experiment_end_date,
+                                                                  directory = forcing_path / "Camels",
+                                                                  basin_id = f"camels_0{HRU_id}"
+                                                                )
+        which gives somthing like:
+
+        .. code-block:: python
+
+            LumpedCaravanForcing(
+                start_time='1997-08-01T00:00:00Z',
+                end_time='2005-09-01T00:00:00Z',
+                directory=PosixPath('/home/davidhaasnoot/eWaterCycle-WSL-WIP/Forcing/Camels'),
+                shape=PosixPath('/home/davidhaasnoot/eWaterCycle-WSL-WIP/Forcing/Camels/shapefiles/camels_01022500.shp'),
+                filenames={
+                    'tasmax': 'camels_01022500_1997-08-01T00:00:00Z_2005-09-01T00:00:00Z_tasmax.nc',
+                    'tasmin': 'camels_01022500_1997-08-01T00:00:00Z_2005-09-01T00:00:00Z_tasmin.nc',
+                    'evspsblpot': 'camels_01022500_1997-08-01T00:00:00Z_2005-09-01T00:00:00Z_evspsblpot.nc',
+                    'pr': 'camels_01022500_1997-08-01T00:00:00Z_2005-09-01T00:00:00Z_pr.nc',
+                    'tas': 'camels_01022500_1997-08-01T00:00:00Z_2005-09-01T00:00:00Z_tas.nc',
+                    'Q': 'camels_01022500_1997-08-01T00:00:00Z_2005-09-01T00:00:00Z_Q.nc'
+                }
+            )
+    """
+
+    @classmethod
+    def retrieve(cls: Type["Caravan"],
```

Comment:
I am in doubt about `retrieve` vs. `generate`...

`generate` would be better for consistency.
`retrieve` is more descriptive of what this method does.

At the least, you perhaps should make `.generate` raise a better error, at the moment you don't overwrite the default forcing's method even though that does contain an implementation.

Reply:
@Peter9192 what is your opinion on this?

Reply:
Made `.generate` now raises an `NotImplementedError` directing you to `retrieve`.
Retrieve makes more sense to me as it also makes it clear to the user it's an already prepared dataset. 
But interested to see what Peter says.

Reply:
From reading the conversation only I think I'm okay with the proposed solution :-)